### PR TITLE
feat: add more hardware information to the link status resources

### PR DESCRIPTION
**Pull request by Unix4ever**
The new fields will be used by the network device selector.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5498)

### CODE REVIEW

File: internal/app/machined/pkg/controllers/network/link_status_test.go [hunk 1]
```diff
@@ -126,6 +128,59 @@ func (suite *LinkStatusSuite) assertNoInterface(id string) error {
 }
 
 func (suite *LinkStatusSuite) TestLoopbackInterface() {
```
Comment (smira) [inline internal/app/machined/pkg/controllers/network/link_status_test.go:130]: feels like test name should be swapped with the next test?

File: pkg/machinery/nethelpers/device.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	}
+
+	if ueventContents == "" {
+		return &DeviceInfo{}, nil
+	}
+
+	device := &DeviceInfo{}
+
+	for _, line := range strings.Split(ueventContents, "\n") {
+		parts := strings.Split(line, "=")
```
Comment (smira) [inline pkg/machinery/nethelpers/device.go:62]: https://pkg.go.dev/strings#Cut is a better way here

**Review by smira — COMMENTED** [review]
can we also add some example output to https://www.talos.dev/v1.0/learn-more/networking-resources/ ?

File: internal/app/machined/pkg/controllers/network/link_status_test.go (hunk 1 above)
Comment (Unix4ever) [reply internal/app/machined/pkg/controllers/network/link_status_test.go:130]: lol, not sure how that happened 🙃

**Review by smira — APPROVED** [review]

**Pull request merged**